### ISSUE TITLE
Update eckit 0.23 -> 1.1.0

### ISSUE THREAD

**Issue by markjolah** [JCSDA/jedi-stack#32]
@srherbener @ytremolet @svahl991 I'll use this zenhub issue as a central tracker for all eckit-1.1.0 upgrade discussions.

I was able to get ufo-bundle to compile with the new eckit.  I made minor mostly `#include` changes to several dozen files in fckit, oops, ufo, and ioda.

I had 47/318 tests fail in ufo-bundle.  I will be going through those failures tomorrow to see what can be done.  I will also check the compilation of fv3-bundle tomorrow.  I have not used mpas-bundle yet, but I suppose I will need to check that too.

Basically I can start to get a more firm estimate of required modification time tomorrow when I see how difficult it is to track down all those failing tests.

**Comment by markjolah** [comment]
We are now also supporting the update to fckit-0.6.3.  Only fckit and oops required changes to accomplish this (so far).  All ufo-bundle tests are passing.

**Comment by markjolah** [comment]
@ytremolet @srherbener @svahl991 Current status: I have gotten tentative approval from most of the major repos.  I suppose the first step will be to update jedi-stack and rebuild the relevant containers, and modules on relevant systems.  Until those are changed, the other repo PRs especially OOPS should not be merged.  Mark M. is on PTO until next week.  I have permissions to update docker hub; currently singularity hub is down for maintenance, and I haven't tried the charliecloud build steps yet.  Also we would need someone with HPC access to update the various machines.

**Comment by srherbener** [comment]
@markjolah, I think Cheyenne is covered by updating jedi-stack. @aerorahul and @danholdaway can help with Discover and Theia.

If singularity hub remains down for a while, we could publish the container on S3.

**Comment by ytremolet** [comment]
If some repos are safe to merge without the new eckit/fckit we can go ahead and merge those.

For the containers, if we are confident we can do it while Mark M is away we can go ahead, otherwise we'll do that when he is back next week. If people start updating eckit on different machines, will it break the system for unaware users? How much do we need to do all that in sync?

**Comment by markjolah** [comment]
I built the docker container last night, and I will test it this afternoon when I have some free time.  I will look into what it takes to build singularity and charliecloud.  

 * I think PRs to ioda, ufo, and fv3-jedi, can be merged as they are backwards compatible.  Travis is getting confused by the OOPS branch with the same name which is not compatible with the current docker image, but they should be OK. 
 * oops, and JCSDA/fckit updates must be coordinated with the jedi-stack and docker updates.  
 * soca has given approval and is ready to merge when the rest of the updates happen.
 * I still have to take some time to get MPAS to build, but changes were trivial and should be backwards compatible.  I just can't confirm that without running the tests.  Maybe I will work on that tomorrow.

**Comment by srherbener** [comment]
Using feature/eckit-1.1.0 branch in ioda (the ioda PR) with all other develop branches passes all tests in ufo-bundle. I will go ahead and merge the ioda PR now.

**Comment by markjolah** [comment]
An updated docker image is now on docker-hub for testing:
```
$ docker pull jcsda/docker:eckit-1.1.0
$ docker run -it jcsda/docker:eckit-1.1.0 -v <path-to-ufo-bundle>:/home/jedi/ufo-bundle
<docker-container> # su - jedi
<docker-container> $ cd ~/ufo-bundle
<docker-container> $ ecbuild -- -H. -B_build
<docker-container> $ cd _build
<docker-container> $ make -j12 && make test
```
